export_trt.py

```diff
@@ -1,32 +1,255 @@
+            torch.cuda.empty_cache()
+
+
+        print(f"Building TensorRT engine for {onnx_path}: {self.engine_path}")
+
+        replaceList = ['"']
+        for match in replaceList:
+            if match in onnx_path:
+                 onnx_path = onnx_path.replace(match,'')
+        p = Profile()
+        if input_profile:
+            for name, dims in input_profile.items():
+                assert len(dims) == 3
+                p.add(name, min=dims[0], opt=dims[1], max=dims[2])
+
+
+
+        config_kwargs = {}
+
+        config_kwargs['preview_features'] = [trt.PreviewFeature.DISABLE_EXTERNAL_TACTIC_SOURCES_FOR_CORE_0805]
+        if enable_preview:
+            # Faster dynamic shapes made optional since it increases engine build time.
+            config_kwargs['preview_features'].append(trt.PreviewFeature.FASTER_DYNAMIC_SHAPES_0805)
+        if workspace_size > 0:
+            config_kwargs['memory_pool_limits'] = {trt.MemoryPoolType.WORKSPACE: workspace_size}
+        if not enable_all_tactics:
+            config_kwargs['tactic_sources'] = []
+
+        engine = engine_from_network(
+            network_from_onnx_path(onnx_path, flags=[trt.OnnxParserFlag.NATIVE_INSTANCENORM]),
+            config=CreateConfig(fp16=fp16,
+                refittable=enable_refit,
+                profiles=[p],
+                load_timing_cache=timing_cache,
+                **config_kwargs
+            ),
+            save_timing_cache=timing_cache
+        )
+        save_engine(engine, path=self.engine_path)
+
+def get_unet_trt_profile(cond_dim, min_bs, opt_bs ,max_bs, min_token_count, opt_token_count, max_token_count, min_width, opt_width, max_width, min_height, opt_height, max_height):
+    profile = {
+            'x' :          ((min_bs * 2, 4, min_height // 8, min_width // 8), (opt_bs * 2, 4, opt_width // 8, opt_height // 8), (max_bs * 2, 4, max_height // 8, max_width // 8)),
+            'timesteps' :  ((min_bs * 2,), (opt_bs * 2,), (max_bs * 2,)),
+            'context' :    ((min_bs * 2, min_token_count // 75 * 77, cond_dim), (opt_bs * 2,  opt_token_count // 75 * 77, cond_dim), (max_bs * 2, max_token_count // 75 * 77, cond_dim)),
+            }
+    return profile
+
+def get_trt_profile_filename(profile):
+    return "unet_w[{},{},{}],h[{},{},{}],b[{},{},{}],t[{},{},{}]" \
+        .format(int(profile['x'][0][3] * 8), int(profile['x'][1][3] * 8),int(profile['x'][2][3] * 8), 
+        int(profile['x'][0][2] * 8), int(profile['x'][1][2] * 8),int(profile['x'][2][2] * 8),
+        int(profile['x'][0][0]/2),int(profile['x'][1][0]/2),int(profile['x'][2][0]/2),
+        int(profile['context'][0][1]/77*75),int(profile['context'][1][1]/77*75),int(profile['context'][2][1]/77*75))
+
+def generate_trt_engine_presets(trt_filename, onnx_filename, profile_512_512_1, profile_512_512_2, profile_512_512_4, profile_768x768x1, profile_768x768x2, profile_768x768x4, use_fp16):
+    
     cond_dim = 768  # XXX should be detected for SD2.0

```

Comment:
This will also break for SDXL right ? Can't we use polygraph to get the inputs ? https://github.com/NVIDIA/TensorRT/blob/release/8.6/tools/Polygraphy/polygraphy/tools/inspect/subtool/model.py#L118-L122